### PR TITLE
[cpu] optimize shape infer of Reshape

### PR DESCRIPTION
**Pull request by tiger100256-hu**
### Details:
 - *optimize the shape infer of Reshape, squeeze, unsqueeze*

### Tickets:
 - *100509*

### CODE REVIEW

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/shape_ops.cpp [hunk 1]
```diff
@@ -84,10 +87,22 @@ class ShapeOpsCPUTest : public testing::WithParamInterface<shapeOpsParams>, virt
...
+                    auto inputData = tensor.data<ov::element_type_traits<ov::element::INT_TYPE>::value_type>(); \
+                    for (size_t j = 0lu; j < data[idx].size(); ++j) { \
+                            inputData[j] =  data[idx][j]; \
+                    } \
+                    break; \
+             }
+                switch (secondInPrc) {
+                    RESHAPE_TEST_CASE(i64)
+                    RESHAPE_TEST_CASE(i32)
+                    RESHAPE_TEST_CASE(i8)
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/shape_ops.cpp:102]: `i8` doesn't make sense here. The second input is a parameter of the operation that is not quantized. Testing `i8` on the second input may only increase the amount of test cases without a practical value.

**Review by maxnick — CHANGES_REQUESTED** [review]

**Review by maxnick — APPROVED** [review]

**Comment by maxnick** [comment]
@tiger100256-hu , please make CI green.

**Comment by tiger100256-hu** [comment]
@maxnick  I make the CI green, I updated below 3 points
1.  fix a pytorch layer test failed issue
      inputShape(1,0) outpattern(-1) is a valid input in the test
2. fix compile issue on windows
     compare between int and uint cause compile issue
3. rebase to master to fix CI issue
    https://github.com/openvinotoolkit/openvino/pull/16573 is merged into master and  fix this issue

**Comment by yuxu42** [comment]
@maxnick @dmitry-gorokhov Whether can it be merged? Thanks!

**Pull request merged**